Core/include/Acts/Utilities/detail/Subspace.hpp

```diff
@@ -204,6 +210,175 @@ class FixedSizeSubspace {
   }
 };
 
+/// Variable-size subspace representation.
+///
+/// @tparam kFullSize Size of the full vector space
+/// @tparam kSize Size of the subspace
+template <std::size_t kFullSize>
+class VariableSizeSubspace {
+  static_assert(kFullSize <= static_cast<std::size_t>(UINT8_MAX),
+                "Full vector space size is larger than the supported range");
+
+  template <typename scalar_t>
+  using ProjectionMatrix = Eigen::Matrix<scalar_t, Eigen::Dynamic, kFullSize>;
+  template <typename scalar_t>
+  using ExpansionMatrix = Eigen::Matrix<scalar_t, kFullSize, Eigen::Dynamic>;
```

Comment:
My one concern here is that we should avoid making the dynamic matrices "viral", so they don't spread to other parts of the code.

I wonder if we should remove or discourage using the projection and expansion matrices completely, and use methods that return statically sized objects instead. What do you think?

Reply:
That's a good point. Removing it or only passing the matrix in as template param might be better I guess? Like `template<MatrixType> fillProjector(MatrixType &matrix)` and then asserting on the dimensions.

Reply:
That would work. Alternatively, you could require supplying an explicit dimension template parameter, although that makes it a bit more annoying.

Reply:
I removed it since it is not used anyways for now